pulsar-functions/utils/src/main/java/org/apache/pulsar/functions/utils/Utils.java

```diff
@@ -331,7 +332,8 @@ public static NarClassLoader extractNarClassLoader(Path archivePath, String pkgU
                     throw new IllegalArgumentException(
                             "Corrupt User PackageFile " + pkgUrl + " with error " + e.getMessage());
                 }
-            } else if (pkgUrl.startsWith("http")) {
+            } else if (pkgUrl.startsWith(org.apache.pulsar.common.functions.Utils.HTTP)
```

Comment:
Why is this necessary?  "startsWith" checks for the prefix "http" which would include "https" no?

Reply:
I have tested https with source and sinks work